packages/common-web/src/arrays.ts

```diff
@@ -1,11 +1,8 @@
-export const keyBy = <T>(arr: T[], key: string): Record<string, T> => {
-  return arr.reduce(
-    (acc, cur) => {
-      acc[cur[key]] = cur
-      return acc
-    },
-    {} as Record<string, T>,
-  )
+export const keyBy = <T>(arr: T[], key: string): Map<string, T> => {
```

Comment:
You also need to change the key type. When using `object[someValue]` in javascript, the key will indeed be casted to string. It is not the case with `Map`s.

```suggestion
export function keyBy<T, K extends keyof T>(arr: readonly T[], key: K): Map<T[K], T> {
```

Note that this is a breaking change, requiring a `minor` version update to the changeset (which is probably fine given the low number of [dependents](https://www.npmjs.com/package/@atproto/common-web?activeTab=dependents)).